packages/core/src/lib/phrase.ts

```diff
@@ -0,0 +1,46 @@
+import { LanguageKey } from '@logto/core-kit';
+import resource from '@logto/phrases-ui';
+import { ResourceLanguage } from '@logto/phrases-ui/lib/types';
+import deepmerge from 'deepmerge';
+
+import { findAllCustomLanguageKeys, findCustomPhraseByLanguageKey } from '@/queries/custom-phrase';
+
+const isBuiltInLanguage = (key: string): key is LanguageKey => Object.keys(resource).includes(key);
+
+const getCustomPhrase = async (key: string | undefined, customLanguages: string[]) =>
+  key && customLanguages.includes(key) ? findCustomPhraseByLanguageKey(key) : {};
+
+const getBuiltInPhrase = (key: string | undefined) =>
+  key && isBuiltInLanguage(key) ? { ...resource[key], languageKey: key } : {};
+
+const has = (object: Record<string, unknown>) => Object.keys(object).length > 0;
+
+export const getPhrase = async (detectedLanguages: string[], fallbackLanguage: string) => {
```

Comment:
Maybe we can set the `fallbackLanguage` type as `LanguageKey`.
Because the fully supported language keys haven't been determined, we can add an issue to track this.

Reply:
Intend to do it later. Added LOG-4221 to track it.

Reply:
Intend to do it later. Added LOG-4221 to track it.